test/Keypad.cpp

```diff
@@ -0,0 +1,31 @@
+#include <Arduino.h>
+#include <ArduinoUnitTests.h>
+
+#include "TankControllerLib.h"
+
+unittest(constructor) {
+  // Test singleton
+  Keypad_TC* singleton1 = nullptr;
+  singleton1 = Keypad_TC::instance();
+  assertNotNull(singleton1);
+  Keypad_TC* singleton2 = nullptr;
+  singleton2 = Keypad_TC::instance();
+  assertNotNull(singleton2);
+  assertEqual(singleton1, singleton2);
+}
+
+unittest(getKey) {
+  Keypad_TC* testKeypad = Keypad_TC::instance();
+  Keypad* puppet = testKeypad->_getPuppet();
```

Comment:
Could we confirm that if there is nothing available that the puppet returns NO_KEY?

Reply:
Testing has been added, but it relies on behavior in the CI version. The tests will pass once the NO_KEY pull request is accepted on that end.